### PR TITLE
feat(cph): add resource cph share app

### PR DESCRIPTION
**Pull request by luoping-12345**
**What this PR does / why we need it**:
add resource cph share app

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```
add resource cph share app
```

## PR Checklist

* [ ] Tests added/passed.

```
make testacc TEST="./huaweicloud/services/acceptance/cph" TESTARGS="-run TestAccShareApp_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cph -v -run TestAccShareApp_basic -timeout 360m -parallel 4
=== RUN   TestAccShareApp_basic
=== PAUSE TestAccShareApp_basic
=== CONT  TestAccShareApp_basic
--- PASS: TestAccShareApp_basic (52.50s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cph       52.564s
```

* [ ] Documentation updated.
* [ ] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

**Comment by Alice1319** [comment]
LGTM

**Comment by yuanxiaojun987** [comment]
LGTM

File: huaweicloud/services/cph/resource_huaweicloud_cph_share_app.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+			},
+			"object_path": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the OBS object path.`,
+			},
+			"pre_install_app": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     0,
```
Comment (houpeng80) [inline huaweicloud/services/cph/resource_huaweicloud_cph_share_app.go]: it is usefull?

File: huaweicloud/services/cph/resource_huaweicloud_cph_share_app.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+		KeepResponseBody: true,
+	}
+
+	deleteShareAppOpt.JSONBody = map[string]interface{}{
+		"package_name": d.Get("package_name"),
+		"server_ids":   []string{d.Get("server_id").(string)},
+	}
+	deleteAdbShareAppResp, err := client.Request("DELETE", deleteShareAppPath, &deleteShareAppOpt)
+	if err != nil {
+		return diag.Errorf("error creating CPH share file: %s", err)
```
Comment (saf3dfsa) [inline huaweicloud/services/cph/resource_huaweicloud_cph_share_app.go]: errorMsg is wrong

**Pull request merged**